addons/binding/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/handler/FroniusBaseThingHandler.java

```diff
@@ -0,0 +1,153 @@
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.fronius.FroniusBridgeConfiguration;
+import org.openhab.binding.fronius.api.ValueUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Basic Handler class for all Fronius services.
+ *
+ * @author Gerrit Beine - Initial contribution
+ * @author Thomas Rokohl - Refactoring to merge the concepts
+ */
+public abstract class FroniusBaseThingHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(FroniusBaseThingHandler.class);
+    private final String serviceDescription;
+    private FroniusBridgeHandler bridgeHandler;
+
+    public FroniusBaseThingHandler(Thing thing) {
+        super(thing);
+        serviceDescription = getDescription();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Command {} is not supported for channel: {}", command, channelUID.getId());
+    }
+
+    @Override
+    public void initialize() {
+        if (getFroniusBridgeHandler() == null) {
+            logger.error("Initializing {} Service is only supported within a bridge", serviceDescription);
+            updateStatus(ThingStatus.OFFLINE);
+            return;
+        }
+        logger.debug("Initializing {} Service", serviceDescription);
+        getFroniusBridgeHandler().registerService(this);
+    }
+
+    private synchronized FroniusBridgeHandler getFroniusBridgeHandler() {
+        if (this.bridgeHandler == null) {
+            Bridge bridge = getBridge();
+            if (bridge == null) {
+                return null;
+            }
+            ThingHandler handler = bridge.getHandler();
+            if (handler instanceof FroniusBridgeHandler) {
+                this.bridgeHandler = (FroniusBridgeHandler) handler;
+            } else {
+                return null;
+            }
+        }
+        return this.bridgeHandler;
+    }
+
+    /**
+     * Update all Channels
+     */
+    protected void updateChannels() {
+        for (Channel channel : getThing().getChannels()) {
+            updateChannel(channel.getUID().getId());
+        }
+    }
+
+    /**
+     * Update the channel from the last data
+     *
+     * @param channelId the id identifying the channel to be updated
+     */
+    protected void updateChannel(String channelId) {
+        if (!isLinked(channelId)) {
+            return;
+        }
+        Object value;
+        value = getValue(channelId);
+
+        State state = null;
+        if (value == null) {
+            state = UnDefType.UNDEF;
```

Comment:
What is that good for? If you do not have a value, you must not send any update, also not an UNDEF.